common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1613,10 +1613,10 @@ def get_course_template_context(self):
         request.user = user
         user_info = cc.User.from_django_user(self.runtime.user).to_dict()
         course_id = self.runtime.course_id
-        course = get_course_with_access(self.runtime.user, 'load_forum', course_id)
+        course = get_course_with_access(self.runtime.user, 'load', course_id, check_if_enrolled=True)
         user_cohort_id = get_cohort_id(user, course_id)
 
-        unsafethreads, query_params = get_threads(request, course_id)
+        unsafethreads, query_params = get_threads(request, course)
```

Comment:
This fake request object has no method, so it's causing a bug for me:

```
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1619, in get_course_template_context
    unsafethreads, query_params = get_threads(request, course)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/newrelic-2.46.0.37/newrelic/api/function_trace.py", line 104, in literal_wrapper
    return wrapped(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/django_comment_client/forum/views.py", line 116, in get_threads
    'group_id': get_group_id_for_comments_service(request, course.id, discussion_id),  # may raise ValueError
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/django_comment_client/utils.py", line 561, in get_group_id_for_comments_service
    if not requested_group_id:
UnboundLocalError: local variable 'requested_group_id' referenced before assignment
```

See `get_group_id_for_comments_service` in `utils.py` and you'll understand why. That code should probably treat either GET or POST as the default.